addons/mass_mailing/models/mailing_trace.py

```diff
@@ -98,7 +98,9 @@ class MailingTrace(models.Model):
         ("mail_bl", "Blacklisted Address"),
         ("mail_optout", "Opted Out"),
         ("mail_dup", "Duplicated Email"),
+        ("mail_bounce", "Email bounced"),
     ], string='Failure type')
+    failure_reason = fields.Text('Failure reason', copy=False)
```

Comment:
No migration script intended (like trace -> message -> notif -> failure reason) ? (would say we don't care about existing traces, maybe only the last few ones, and not sure it deserves a migration script).